include/lsToDiskMesh.hpp

```diff
@@ -21,30 +21,38 @@ template <class T, int D, class N = T> class lsToDiskMesh {
   typedef typename lsDomain<T, D>::DomainType hrleDomainType;
   typedef std::unordered_map<unsigned long, unsigned long> translatorType;
 
-  lsSmartPointer<lsDomain<T, D>> levelSet = nullptr;
+  std::vector<lsSmartPointer<lsDomain<T, D>>> levelSets;
   lsSmartPointer<lsMesh<N>> mesh = nullptr;
   lsSmartPointer<translatorType> translator = nullptr;
   T maxValue = 0.5;
   bool buildTranslator = false;
+  static constexpr double wrappingLayerEpsilon = 1e-4;
 
 public:
   lsToDiskMesh() {}
 
+  lsToDiskMesh(lsSmartPointer<lsMesh<N>> passedMesh, T passedMaxValue = 0.5)
+      : mesh(passedMesh), maxValue(passedMaxValue) {}
+
   lsToDiskMesh(lsSmartPointer<lsDomain<T, D>> passedLevelSet,
                lsSmartPointer<lsMesh<N>> passedMesh, T passedMaxValue = 0.5)
-      : levelSet(passedLevelSet), mesh(passedMesh), maxValue(passedMaxValue) {}
+      : mesh(passedMesh), maxValue(passedMaxValue) {
+    levelSets.push_back(passedLevelSet);
+  }
 
   lsToDiskMesh(lsSmartPointer<lsDomain<T, D>> passedLevelSet,
                lsSmartPointer<lsMesh<N>> passedMesh,
                lsSmartPointer<translatorType> passedTranslator,
                T passedMaxValue = 0.5)
-      : levelSet(passedLevelSet), mesh(passedMesh),
-        translator(passedTranslator), maxValue(passedMaxValue) {
+      : mesh(passedMesh), translator(passedTranslator),
+        maxValue(passedMaxValue) {
+    levelSets.push_back(passedLevelSet);
     buildTranslator = true;
   }
 
-  void setLevelSet(lsSmartPointer<lsDomain<T, D>> passedLevelSet) {
```

Comment:
This function should stay here for backwards compatibility!
Adding "insertNextLevelSet" is still a great idea, though ;)

Reply:
Since the effect of "setLevelSet" and "insertNextLevelSet" would then be exactly the same, should I remove the "insertNextLevelSet" or leave it in for a maybe clearer interface?

Reply:
Leave it in so people will only use it, and in a couple versions time we can deprecate and remove "setLevelSet"